File: components/alert-status/flutter/ciata_alert_status.dart
import 'package:flutter/material.dart';

/// Implementação experimental do CMP-0007 Alert/Status para Flutter.
class CiataAlertStatus extends StatelessWidget {
  CiataAlertStatus({
    super.key,
    required String message,
    String? title,
    String priority = 'status',
  })  : message = message.trim(),
        title = title?.trim(),
        priority = priority.trim() {
    if (this.message.isEmpty) {
      throw ArgumentError.value(message, 'message', 'não pode ser vazio');
    }
    if (this.priority != 'status' && this.priority != 'alert') {
      throw ArgumentError.value(
        priority,
        'priority',
        'deve ser status ou alert',
      );
    }
  }

  final String message;
  final String? title;
  final String priority;

  @override
  Widget build(BuildContext context) {
    final normalizedTitle = title?.isNotEmpty == true ? title : null;
    final text = [
      if (normalizedTitle != null) normalizedTitle,
      message,
    ].join('. ');

    return Semantics(
      liveRegion: true,
      label: text,
      child: ExcludeSemantics(
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            if (normalizedTitle != null)
              Text(
                normalizedTitle,
                style: Theme.of(context).textTheme.titleMedium,
              ),
            Text(message),
          ],
        ),
      ),
    );
  }
}

File: components/alert-status/android/CiataAlertStatus.kt
package org.ciata.ds.alertstatus

import androidx.compose.foundation.layout.Column
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.LiveRegionMode
import androidx.compose.ui.semantics.liveRegion
import androidx.compose.ui.semantics.semantics

/** Implementação experimental do CMP-0007 Alert/Status para Jetpack Compose. */
@Composable
fun CiataAlertStatus(
    message: String,
    modifier: Modifier = Modifier,
    title: String? = null,
    priority: String = "status",
) {
    val normalizedMessage = message.trim()
    val normalizedTitle = title?.trim()?.takeIf { it.isNotEmpty() }
    val normalizedPriority = priority.trim()

    require(normalizedMessage.isNotEmpty()) { "message não pode ser vazio." }
    require(normalizedPriority == "status" || normalizedPriority == "alert") {
        "priority deve ser status ou alert."
    }

    val region = if (normalizedPriority == "alert") {
        LiveRegionMode.Assertive
    } else {
        LiveRegionMode.Polite
    }

    Column(
        modifier = modifier.semantics {
            liveRegion = region
        },
    ) {
        normalizedTitle?.let { Text(it) }
        Text(normalizedMessage)
    }
}
